app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -142,4 +142,18 @@ def available_vlans_and_hosts(options = {})
     end
     return vlans, hosts
   end
+
+  def allowed_storage_profiles(_options = {})
+    return [] if (src = resources_for_ui).blank? || src[:vm].nil?
+    @filters[:StorageProfile] ||= begin
+      template = load_ar_obj(src[:vm])
+      @values[:placement_storage_profile] = [template.storage_profile.try(:id), template.storage_profile.try(:name)]
```

Comment:
In regards to RBAC - There may be an instance where a template only has 1 storage_profile, but RBAC doesn't allow the user to see that profile.  We'd probably want to handle that case as well.